src/reactionsystem.jl

```diff
@@ -181,7 +186,27 @@ function Reaction(rate, subs, prods, substoich, prodstoich;
         convert.(stoich_type, netstoich) : netstoich
     end
 
-    Reaction(value(rate), subs, prods, substoich′, prodstoich′, ns, only_use_rate)
+    # Check that all metadata entries are unique. (cannot use `in` since some entries may be symbolics).
+    if any(any(isequal(metadata[i][1], entry[1]) for entry in metadata[i+1:end]) 
+                                                 for i in eachindex(metadata))
+        error("Repeated entries for the same metadata encountered in the following metadata set: $([entry[1] for entry in metadata]).")
+    end
+
+    # Deletes potential `:only_use_rate => ` entries from the metadata.
+    if any(:only_use_rate == entry[1] for entry in metadata) 
+        deleteat!(metadata, findfirst(:only_use_rate == entry[1] for entry in metadata))
+    end
+
+    # Ensures metadata have the correct type.
+    metadata = convert(Vector{Pair{Symbol, Any}}, metadata)
+
+    Reaction(value(rate), subs, prods, substoich′, prodstoich′, ns, only_use_rate, metadata)
+end
+
+# Checks if a metadata input has an entry :only_use_rate => true
+function metadata_only_use_rate_check(metadata)
+    any(:only_use_rate == entry[1] for entry in metadata) || (return false)
+    return Bool(metadata[findfirst(:only_use_rate == entry[1] for entry in metadata)][2])
```

Comment:
Why not just use `findfirst` and if it returns `nothing` you know that it isn't defined? You shouldn't need two searches through the `metadata`.

Reply:
yes, that is better, fixed.